instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -0,0 +1,130 @@
+            boolean_literals: /\b(?:true|false|null)\b/i,
+            hexadecimal_literals: /0x[0-9a-fA-F]+/,
+            comments: /(?:#|--).*?(?=\r|\n|$)/i,
+            multi_line_comments: %r{\/\*(?:[^\/]|\/[^*])*?(?:\*\/|\/\*.*)}
+          }.freeze
+
+          MYSQL_COMPONENTS = %i[
+            single_quotes
+            double_quotes
+            numeric_literals
+            boolean_literals
+            hexadecimal_literals
+            comments
+            multi_line_comments
+          ].freeze
+
+          FULL_SQL_REGEXP = Regexp.union(MYSQL_COMPONENTS.map { |component| COMPONENTS_REGEX_MAP[component] })
+
+          def query(sql)
+            attributes = client_attributes
+
+            case config[:db_statement]
+            when :obfuscate
+              attributes[::OpenTelemetry::SemanticConventions::Trace::DB_STATEMENT] = obfuscate_sql(sql)
+            when :include
+              attributes[::OpenTelemetry::SemanticConventions::Trace::DB_STATEMENT] = sql
+            end
+
+            tracer.in_span(
+              database_span_name(sql),
+              attributes: attributes,
+              kind: :client
+            ) do
+              super(sql)
+            end
+          end
+
+          private
+
+          def obfuscate_sql(sql)
+            if sql.size > 2000
+              'SQL query too large to remove sensitive data ...'
+            else
+              obfuscated = sql.gsub(FULL_SQL_REGEXP, '?')
+              obfuscated = 'Failed to obfuscate SQL query - quote characters remained after obfuscation' if detect_unmatched_pairs(obfuscated)
+              obfuscated
+            end
+          end
+
+          def detect_unmatched_pairs(obfuscated)
+            # We use this to check whether the query contains any quote characters
+            # after obfuscation. If so, that's a good indication that the original
+            # query was malformed, and so our obfuscation can't reliably find
+            # literals. In such a case, we'll replace the entire query with a
+            # placeholder.
+            %r{'|"|\/\*|\*\/}.match(obfuscated)
+          end
+
+          def database_span_name(sql)
+            # Setting span name to the SQL query without obfuscation would
+            # result in PII + cardinality issues.
+            # First attempt to infer the statement type then fallback to
+            # current Otel approach {database.component_name}.{database_instance_name}
+            # https://github.com/open-telemetry/opentelemetry-python/blob/39fa078312e6f41c403aa8cad1868264011f7546/ext/opentelemetry-ext-dbapi/tests/test_dbapi_integration.py#L53
+            # This creates span names like mysql.default, mysql.replica, postgresql.staging etc.
+
+            statement_type = extract_statement_type(sql)
+
+            return statement_type unless statement_type.nil?
+
+            # fallback
+            'mysql'
+          end
+
+          def client_attributes
```

Comment:
small nit, would it be possible to support the `peer_service` attribute via a  `config[:peer_service]` option here? Just so there's parity with other db instrumentations? https://github.com/open-telemetry/opentelemetry-ruby/blob/4216a55be5ad224100f6c2802e8ec5fc04f1e316/instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb#L129

Reply:
Sure thing I will add it. 

I left this off because I do not use this attribute in any of our instrumentations. Setting it as an instrumentation options makes it difficult to use on a per connection instance since unrelated instances end up getting the same attribute and there isn't an easy way to override it AFAICT.

Reply:
hmm, let's skip it then. I think we bumped into similar attribution issues for `peer.service` with kafka related instrumentation, and we chose to punt on those.

Reply:
it's all good. I added it anyway.